tests/dispense.rs

```diff
@@ -255,8 +278,57 @@ async fn many_concurrent_requests() {
         .collect::<Vec<_>>();
 
     assert_eq!(COUNT, txs.len());
-    assert_eq!(
-        test_balance,
-        COUNT as u64 * context.faucet_config.dispense_amount
-    );
+}
+
+#[tokio::test]
+async fn dispense_once_per_day() {
+    let mut rng = StdRng::seed_from_u64(42);
+    let recipient_address: Address = rng.gen();
+    let recipient_address_str = format!("{}", &recipient_address);
+    let context = TestContext::new(rng).await;
+    let addr = context.addr;
+
+    let dispense_interval = 24 * 60 * 60;
+    let time_increment = dispense_interval / 6;
+
+    let response = reqwest::Client::new()
+        .post(format!("http://{addr}/dispense"))
+        .json(&json!({
+            "captcha": "",
+            "address": recipient_address_str.clone(),
+        }))
+        .send()
+        .await
+        .expect("First dispensing request should be successful");
+
+    assert_eq!(response.status(), reqwest::StatusCode::CREATED);
+
+    for _ in 0..5 {
+        context.clock.advance(time_increment);
+
+        let response = reqwest::Client::new()
+            .post(format!("http://{addr}/dispense"))
+            .json(&json!({
+                "captcha": "",
+                "address": recipient_address_str.clone(),
+            }))
+            .send()
+            .await
+            .expect("Subsequent dispensing requests should be successful with status 500");
```

Comment:
Nit: I think the `expect` should probably say 
```suggestion
            .expect("Subsequent dispensing requests should be successfully sent");
```
since this expect is just checking "if there was an error while sending request, redirect loop was detected or redirect limit was exhausted" (https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.send)